### PR TITLE
test: Refining E2E Testing with API Mocking

### PR DESCRIPTION
**Pull request by Andepande**
## **Description**
This change enhances the mock functionality to enable multiple dynamic events to be passed through. This will evolve over time with how people use it. A corresponding document has been created in the contributors' repository.

## **Related issues**

Fixes:
https://github.com/orgs/MetaMask/projects/60/views/3?pane=issue&itemId=81125053&issue=MetaMask%7Cmobile-planning%7C1944

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by SamuelSalas — DISMISSED** [review]
LGTM

File: e2e/mockServer/mock-server.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-console */
```
Comment (cortisiko) [inline e2e/mockServer/mock-server.js:1]: You have the file and directory names inverted. The file name should be `mockServer.js` and the name of the directory should be` mock-server`

File: e2e/specs/mock-specs/suggestedGasApi.mock.spec.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import ImportAccountView from '../../pages/importAccount/ImportAccountView.js';
+import Accounts from '../../../wdio/helpers/Accounts.js';
+import { withFixtures } from '../../fixtures/fixture-helper.js';
+import FixtureBuilder from '../../fixtures/fixture-builder.js';
+import TestHelpers from '../../helpers.js';
+import SuccessImportAccountView from '../../pages/importAccount/SuccessImportAccountView.js';
+import { mockEvents } from '../../mockServer/mock-config/mock-events.js';
+
+describe(
+  SmokeCore(
```
Comment (cortisiko) [inline e2e/specs/mock-specs/suggestedGasApi.mock.spec.js:27]: This test seems like a good candidate for regression not smoke

File: e2e/mockServer/mock-responses/mock-responses.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (cortisiko) [inline e2e/mockServer/mock-responses/mock-responses.json:1]: We should create individual JSONs for the mock endpoints; therefore, the names of the response data should match those. In this case, the name of this file should be `gas-api-responses.json`. The name we currently have implies that we are going to have an exhaustive JSON file for all the mock responses.

**Review by cortisiko — CHANGES_REQUESTED** [review]
Just some small feedback; otherwise, things are looking good.

File: e2e/api-mocking/mock-config/mock-events.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import {
+  suggestedGasApiResponses,
+  suggestedGasFeesApiGanache,
+} from '../mock-responses/gas-api-responses.json';
+
```
Comment (cortisiko) [inline e2e/api-mocking/mock-config/mock-events.js:5]: Can you add JSDocs here as well?

File: e2e/api-mocking/mock-config/mockUrlCollection.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "defaultMockPort": 8000
```
[the remark this answers is not in the document]
Comment (cortisiko) [reply e2e/api-mocking/mock-config/mockUrlCollection.json:2]: Okay so you defined The mock server port two places, here and then again here: https://github.com/MetaMask/metamask-mobile/pull/11797/files#diff-03b7a8cfcc1c4512beea1ccb712a7c6570c734f7e2e359ade57c65c5ffeb17a4R6 

I do not see a need for having it defined two places. If it is used in the shims you can just import the default port from the mock server file

**Review by cortisiko — APPROVED** [review]

**Pull request merged**